packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+    const errorInvalidResponse = 'Invalid response: Received more images than the specified max limit in the response.';
+    /**
+     * Capture one or multiple image(s) using camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than 10 or lesser than 1.
+     */
+    export async function captureImages(cameraImageInputs: CameraImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (
+        !cameraImageInputs ||
+        cameraImageInputs.maxVisualMediaCount > 10 ||
+        cameraImageInputs.maxVisualMediaCount < 1
+      ) {
+        throw errorInvalidCount;
+      }
+      if (cameraImageInputs.sourceProps.source != Source.Camera) {
+        throw errorInvalidSource;
+      }
+      // waiting the response from hub-SDK
+      const files = await sendAndHandleSdkError<VisualMediaFile[]>(
+        'visualMedia.image.captureImages',
+        cameraImageInputs,
+      );
+      // to ensure the number of images in the response is within the maximum limit.
+      if (files.length > cameraImageInputs.maxVisualMediaCount) {
+        throw errorInvalidResponse;
+      }
+      return files;
+    }
+
+    /**
+     * Upload the existing image(s) from the gallery.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than 10 or lesser than 1.
+     */
+    export async function retrieveImages(galleryImageInputs: GalleryImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (
+        !galleryImageInputs ||
+        galleryImageInputs.maxVisualMediaCount > 10 ||
+        galleryImageInputs.maxVisualMediaCount < 1
+      ) {
+        throw errorInvalidCount;
+      }
+      if (galleryImageInputs.sourceProps.source != Source.Gallery) {
+        throw errorInvalidSource;
+      }
+      // waiting the response from hub-SDK
+      const files = await sendAndHandleSdkError<VisualMediaFile[]>(
+        'visualMedia.image.retrieveImages',
+        galleryImageInputs,
+      );
+      // to ensure the number of images in the response is within the maximum limit.
+      if (files.length > galleryImageInputs.maxVisualMediaCount) {
```

Comment:
Similar to above comment. Why are we throwing error when we get valid response ?

Reply:
Same to the above comment, and I will make sure I mentioned your point to our PM for future improvement.